crates/transaction-pool/src/lib.rs

```diff
@@ -415,25 +416,16 @@ where
         &self,
         tx_hashes: Vec<TxHash>,
         limit: GetPooledTransactionLimit,
-    ) -> Vec<<<V as TransactionValidator>::Transaction as PoolTransaction>::Pooled> {
-        self.pool.get_pooled_transaction_elements(tx_hashes, limit)
-    }
-
-    fn get_pooled_transactions_as<P>(
-        &self,
-        tx_hashes: Vec<TxHash>,
-        limit: GetPooledTransactionLimit,
-    ) -> Vec<P>
-    where
-        <Self::Transaction as PoolTransaction>::Pooled: Into<P>,
+    ) -> Vec<RecoveredTx<<<V as TransactionValidator>::Transaction as PoolTransaction>::Pooled>>
     {
-        self.pool.get_pooled_transactions_as(tx_hashes, limit)
```

Comment:
this existed only so we can avoid a double-collect,
we can either find a way to enforce the 
`From<RecoveredTx<N::PooledTransaction>>` again which would be a bit weird because this would always be into_signed.
or we add another helper fn to this trait that strips the `Recovered` layer away and returns just the pooled type

Reply:
looks like this fn is only used here, so I've just made it return transactions with already stripped signer e6c0f18

Reply:
makes sense